pkg/kubelet/kubelet_node_status.go

```diff
@@ -308,17 +307,7 @@ func (kl *Kubelet) initialNode(ctx context.Context) (*v1.Node, error) {
 		node.Labels[label] = value
 	}
 
-	nodeTaints := make([]v1.Taint, 0)
-	if len(kl.registerWithTaints) > 0 {
-		taints := make([]v1.Taint, len(kl.registerWithTaints))
-		for i := range kl.registerWithTaints {
-			if err := k8s_api_v1.Convert_core_Taint_To_v1_Taint(&kl.registerWithTaints[i], &taints[i], nil); err != nil {
-				return nil, err
-			}
-		}
-		nodeTaints = append(nodeTaints, taints...)
-	}
-
+	nodeTaints := kl.registerWithTaints
```

Comment:
because we modify `nodeTaints` later in the function, create a copy of the list here instead of assigning `kl.registerWithTaints` directly

Reply:
Done